src/go/rpk/pkg/cli/cmd/debug/bundle/bundle_k8s_linux.go

```diff
@@ -72,3 +80,164 @@ func executeK8SBundle(ctx context.Context, bp bundleParams) error {
 	fmt.Printf("Debug bundle saved to %q\n", f.Name())
 	return nil
 }
+
+// adminAddressesFromK8S returns the admin API host:port list by querying the
+// K8S Api.
+func adminAddressesFromK8S(ctx context.Context) ([]string, error) {
+	// This is intended to run only in a k8s cluster:
+	k8sCfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get kubernetes cluster configuration: %v", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(k8sCfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create kubernetes client: %v", err)
+	}
+
+	// Get pods in the 'redpanda' namespace.
+	pods, err := clientset.CoreV1().Pods("redpanda").List(ctx, v1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("unable to get kubernetes pods: %v", err)
+	}
+
+	// Get the admin addresses from ContainerPort.
+	var adminAddresses []string
+	for _, p := range pods.Items {
+		for _, port := range p.Spec.Containers[0].Ports {
+			if port.Name == "admin" {
+				a := fmt.Sprintf("%v:%v", p.Status.PodIP, port.ContainerPort)
+				adminAddresses = append(adminAddresses, a)
+			}
+		}
+	}
+
+	return adminAddresses, nil
+}
+
+// saveClusterAdminAPICalls save the following admin API request to the zip:
+//  - Cluster Health: /v1/cluster/health_overview
+//  - Brokers: /v1/brokers
+//  - License Info: /v1/features/license
+//  - Cluster Config: /v1/cluster_config
+//  - Reconfigurations: /v1/partitions/reconfigurations
+func saveClusterAdminAPICalls(ctx context.Context, ps *stepParams, fs afero.Fs, cfg *config.Config) step {
+	return func() error {
+		adminAddresses, err := adminAddressesFromK8S(ctx)
```

Comment:
Should adminAddressesFromK8s be done once in the run func, and then each address passed to both these functions?